### PR TITLE
Updating the vault injector connectivity docs

### PR DESCRIPTION
**Pull request by tvoran**
Adding more detail about connectivity requirements, noting that
masters sometimes need to connect to workers on :8080, and
considerations when Vault is running outside of Kubernetes.

Related to https://github.com/hashicorp/vault-helm/issues/366

### CODE REVIEW

File: website/pages/docs/platform/k8s/injector/examples.mdx [hunk 1]
```diff
@@ -19,12 +19,17 @@ using higher level constructs such as deployments, jobs or statefulsets.
 Before applying Vault Agent injection annotations to pods, the following requirements
 should be satisfied:
 
-- the master Kubernetes nodes can receive traffic from worker nodes on port `8080`,
+- the master Kubernetes nodes can connect to worker nodes on port `8080`,
```
Comment (calvn) [inline website/pages/docs/platform/k8s/injector/examples.mdx:22]: I like this, definitely more clear!

Comment (calvn) [reply website/pages/docs/platform/k8s/injector/examples.mdx:22]: This is a two-way policy setup right, i.e. the master nodes needs ingress policy from the worker nodes and the workers needs egress policy to the master nodes?

**Review by tomhjp — APPROVED** [review]
Really nice clarification 👍

**Review by jasonodonnell — APPROVED** [review]
LGTM!

**Pull request merged**